website/content/posts/introducing-action-validate-image.md

```diff
@@ -0,0 +1,119 @@
+We employ SLSA tooling to generate provenance. This helps track the build process and adds traceability to the container image.
+```yaml
+- name: slsa-github-generator
+  uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.9.0 
+  with:
+    image: ${{ needs.build.outputs.image }}
+    digest: ${{ needs.build.outputs.digest }}
+    registry-username: ${{ github.actor }}
+  secrets:
+    registry-password: ${{ secrets.GITHUB_TOKEN }}
+```
+
+### EC Action Validate
+EC Action Validate works by assessing your container images against a set of validation checks. You can customize these checks through a policy to align with the specific security and compliance guidelines of your organization or industry. Whether the image passes or fails, you'll receive a GitHub summary output, and additional logs will be available in YAML format within the action.
+```yaml
+- name: Validate image (keyless)
+  uses: enterprise-contract/action-validate-image@v1.0.31 
+  with:
+    image: ${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }}
+    identity: https:\/\/github\.com\/(slsa-framework\/slsa-github-generator|${{ github.repository_owner }}\/${{ github.event.repository.name }})\/
+    issuer: https://token.actions.githubusercontent.com
+    policy: github.com/enterprise-contract/config//github-default
+```
+- **`image`**: Specifies the container image to be validated. It uses the image and digest output from a previous build step in the workflow.
+
+- **`identity`**: Defines the identity or entities that are allowed to perform the validation. In this case, it's either from the SLSA GitHub Generator or the repository owner.
+
+- **`issuer`**: Specifies the issuer of the token used for authentication. Here, it's set to GitHub's token actions issuer URL.
+
+- **`policy`**: Points to the policy configuration to use for validation checks. The policy can be either [predefined](https://github.com/enterprise-contract/config) or custom and is stored in a specific repository.
+
+
+### Promote Image
+Upon successful validation, this step promotes the image by pushing the latest validated tag. This ensures that only images that have passed checks are promoted. 
+```yaml
+- name: Push latest-validated image tag 
+  run: |
+      skopeo copy --dest-creds=${{ github.actor }}:${{ github.token }}       
+      docker://${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }}      
+      docker://${{ needs.build.outputs.image }}:latest
+```
+
+To see the full workflow vist [here](https://github.com/enterprise-contract/golden-container/blob/main/.github/workflows/release.yaml)
+
+## Example of Long-Lived Public-Key Authentication
+Here is a verison of the EC Action Validate that uses long-lived public-key authentication. This method uses a public key to authenticate the image, which is stored in a GitHub repository. This key is then used to verify the image's signature and ensure its integrity.
```

Comment:
"authentication" terminology here is not right.

Reply:
```suggestion
Here is a verison of the EC Action Validate that uses long-lived. This method uses a public key, stored in a secret variable, to verify the image signature, thereby ensuring its integrity through a three-stage validation process: Signature Verification, Attestation Verification, and Policy Conformance which we have usen `slsa3`.
```